File: src/types/sale.ts
export type SaleStatus = "concluida" | "cancelada";

export interface SaleItem {
  productId: string;
  productName: string;
  quantity: number;
  unitPrice: number;
  subtotal: number;
}

export interface Sale {
  id: string;
  companyId: string;
  date: string;
  status: SaleStatus;
  items: SaleItem[];
  total: number;
}

export interface ApiSale {
  id: string;
  tenant_id: string;
  total_amount: number | string;
  sale_date: string;
  created_by: string;
  created_at: string;
}

export interface ApiSaleItem {
  id: string;
  sale_id: string;
  product_id: string;
  product_name: string;
  quantity: number;
  unit_price: number | string;
  subtotal: number | string;
}

export interface ApiSaleDetail extends ApiSale {
  items: ApiSaleItem[];
}

export interface SalePayloadItem {
  productId: string;
  quantity: number;
}

export interface SalePayload {
  date: string;
  items: SalePayloadItem[];
}

export interface SaleFilters {
  startDate?: string;
  endDate?: string;
}

File: backend/app/schemas/sale.py
from datetime import datetime
from decimal import Decimal

from pydantic import BaseModel, ConfigDict, Field


class SaleCreateItem(BaseModel):
    product_id: str
    quantity: int = Field(gt=0)


class SaleCreate(BaseModel):
    sale_date: datetime
    items: list[SaleCreateItem] = Field(min_length=1)


class SaleItemResponse(BaseModel):
    model_config = ConfigDict(from_attributes=True)

    id: str
    sale_id: str
    product_id: str
    quantity: int
    unit_price: Decimal
    subtotal: Decimal


class SaleResponse(BaseModel):
    model_config = ConfigDict(from_attributes=True)

    id: str
    tenant_id: str
    total_amount: Decimal
    sale_date: datetime
    created_by: str
    created_at: datetime


class SaleDetailResponse(SaleResponse):
    items: list[SaleItemResponse]
